Source/DafnyTestGeneration/TestMethod.cs

```diff
@@ -450,7 +470,9 @@ public class TestMethod {
                  immutableFields) {
           constFieldValues.Add(GetFieldValue(field, variable));
         }
-        ValueCreation.Add(new(varId, dafnyType, $"{GetSynthesizeMethodName(dafnyType.ToString())}({string.Join(", ", constFieldValues)})"));
+        // TODO: How should we handle synthesized method names here?
+        // ValueCreation.Add(new(varId, dafnyType, $"{GetSynthesizeMethodName(dafnyType.ToString())}({string.Join(", ", constFieldValues)})"));
+        ValueCreation.Add(new(varId, dafnyType, $"new {dafnyType.ToString()}({string.Join(", ", constFieldValues)})"));
```

Comment:
Is there a guarantee there is a constructor that takes such parameters? If not, I don't think we would be able to merge this into main Dafny. Instead, perhaps we can add an option that effectively gets rid of synthesize methods and assumes there is always an appropriate constructor?

Reply:
Yeah, this is one of the main hacks of this patch. A more general solution needs to be worked out.